### PR TITLE
Switch tests to use new declaration action UI, add new testcase

### PR DESCRIPTION
**Pull request by cibelius**
## Description

* Switch tests to use new declaration action menu
* Add new test case: 'Approval required for late registration' -flag blocks direct registration

## Checklist

- [x] I have linked the correct Github issue under "Development"
- [x] I have tested the changes locally, and written appropriate tests
- [x] I have tested beyond the happy path (e.g. edge cases, failure paths)
- [ ] I have updated the changelog with this change (if applicable)
- [x] I have updated the GitHub issue status accordingly

### CODE REVIEW

File: e2e/testcases/birth/declarations/birth-declaration-2.spec.ts [hunk 1]
```diff
@@ -535,6 +536,17 @@ test.describe.serial('2. Birth declaration case - 2', () => {
...
 
+    test('2.1.6.1 Validate declare action not available before filling in signature and comment', async () => {
+      await page.getByRole('button', { name: 'Action' }).click()
+      const declareButton = page.getByText('Declare', { exact: true })
+      await expect(declareButton).toBeVisible()
+      await expect(declareButton).toHaveAttribute('disabled')
+
+      const notifyButton = page.getByText('Notify', { exact: true })
+      await expect(notifyButton).toBeVisible()
+      await expect(notifyButton).not.toHaveAttribute('disabled')
```
Comment (Nil20) [inline e2e/testcases/birth/declarations/birth-declaration-2.spec.ts]: Couldn't we use the `selectDeclarationAction` helper?

File: e2e/testcases/birth/8-validate-declaration-review-page.spec.ts [hunk 2]
```diff
@@ -893,18 +892,6 @@ test.describe.serial('8. Validate declaration review page', () => {
...
-      await page.getByRole('button', { name: 'Confirm' }).click()
-      await ensureOutboxIsEmpty(page)
-      await page.getByText('Sent for approval').click()
-
-      await expect(
-        page.getByRole('button', {
-          name: formatName(declaration.child.name)
-        })
-      ).toBeVisible()
-    })
```
Comment (Nil20) [inline e2e/testcases/birth/8-validate-declaration-review-page.spec.ts:907]: Wouldn't the declaration have to be sent for approval? Is it intentional?

File: e2e/testcases/custom-actions-and-flags/approval-of-late-birth-registration.spec.ts [hunk 3]
```diff
@@ -288,7 +295,7 @@ test.describe('Birth with non-late registration will not have flag or Approve-ac
 
       await page.getByPlaceholder('dd').fill('25')
       await page.getByPlaceholder('mm').fill('11')
-      await page.getByPlaceholder('yyyy').fill('2025')
```
Comment (Nil20) [inline e2e/testcases/custom-actions-and-flags/approval-of-late-birth-registration.spec.ts:291]: `mother.dob` accepted birth year as 2025?

File: e2e/helpers.ts [hunk 4]
```diff
@@ -587,3 +587,35 @@ export async function expectRowValueWithChangeButton(
...
+  } else {
+    await expect(actionButton).toHaveAttribute('disabled')
+  }
+
+  await page.getByRole('button', { name: 'Action', exact: true }).click()
+}
+
+export async function selectDeclarationAction(
+  page: Page,
+  action: string,
```
Comment (Nil20) [inline e2e/helpers.ts]: Very useful helper.

Could be a little better if the `action` types are real actions rather than plain string.
```suggestion
  action: 'Notify' | 'Declare' | 'Validate',
```

File: e2e/testcases/advanced-search/1-search-birth-event-declaration-registration-details.spec.ts [hunk 5]
```diff
@@ -22,12 +22,9 @@ test.describe
 
     await createDeclaration(token, {
       'child.dob': faker.date
-        // Randomly chosen DOB between 2010-01-01 and 2020-12-31
-        // Ensures the created record appears on the first page of search results
-        .between({ from: '2010-01-01', to: '2020-12-31' })
+        .between({ from: '2025-09-10', to: '2025-11-28' })
```
Comment (Nil20) [inline e2e/testcases/advanced-search/1-search-birth-event-declaration-registration-details.spec.ts:25]: Is there no test for late registered records? Test data suggest there could have been one. Ignore, if there's none.

**Review by Nil20 — APPROVED** [review]
Great refactor with the action menu helpers. They should be useful going forward. 💯
